python-threatexchange/threatexchange/stopncii/api.py

```diff
@@ -0,0 +1,251 @@
+        with self._get_session() as session:
+            response = session.post(url, json=json, params=params or {})
+            response.raise_for_status()
+            return response.json()
+
+    def fetch_hashes(
+        self, *, start_timestamp: int = None, next_page: str = None
+    ) -> FetchHashesResponse:
+        """
+        Fetch a series of update records from the hash API.
+
+        Records represent the current snapshot of all data, so if you see
+        the same SignalType+Hash in a later iteration, it should completely
+        replace the previously observed record.
+        """
+        if start_timestamp is None:
+            start_timestamp = self.DEFAULT_START_TIMESTAMP
+        params: t.Dict[str, t.Any] = {"startTimestamp": start_timestamp}
+        if next_page is not None:
+            params["nextPageToken"] = next_page
+        json_val = self._get("FetchHashes", **params)
+        logging.debug("StopNCII FetchHashes returns: %s", json_val)
+        return dacite.from_dict(
+            data_class=FetchHashesResponse,
+            data=json_val,
+            config=dacite.Config(cast=[enum.Enum, set]),
+        )
+
+    def fetch_hashes_iter(
+        self, start_timestamp: int = None
+    ) -> t.Iterator[FetchHashesResponse]:
+        """
+        A simple wrapper around FetchHashes to keep fetching until complete.
+
+        You could get the entire record stream from:
+        everything = {
+            (record.signalType, record.hashValue): record
+            for result in api.fetch_hashes_iter()
+            for record in result.hashRecords
+        }
+
+        """
+        has_more = True
+        next_page = None
+        while has_more:
+            result = self.fetch_hashes(
+                start_timestamp=start_timestamp, next_page=next_page
+            )
+            has_more = result.hasMoreRecords
+            next_page = result.nextPageToken
+            yield result
+
+    def submit_feedback(
+        self,
+        signal_type: StopNCIISignalType,
+        hash_value: str,
+        feedback: StopNCIICSPFeedback,
+    ) -> None:
+        """
+        Convenience accessor for submitting feedback for a single hash.
+
+        Note that even a single hash may correspond to multiple cases.
+        """
+        self.submit_feedbacks({(signal_type, hash_value): feedback})
+
+    def submit_feedbacks(
+        self, feedbacks: t.Dict[t.Tuple[StopNCIISignalType, str], StopNCIICSPFeedback]
+    ) -> None:
+        """Submit feedback on multiple hashes"""
+        now = int(time.time())
+        # Now for the fun hacky part, convert to expected format
+        records = []
+        for (signal_type, hashValue), feedback in feedbacks.items():
+            fb_dict = asdict(feedback)
+            fb_dict["tags"] = list(fb_dict["tags"])  # Convert from set
+            fb_dict["feedbackValue"] = str(
+                fb_dict["feedbackValue"]
+            )  # Convert from enum
```

Comment:
Can't this be done in an instance method on the `StopNCIICSPFeedback` class?

If the behavior were to change, tracking this usage down might be easy to miss.